File: erlang/p_04.erl
-module(p_04).
-export([my_length/1]).

my_length(L) ->
    my_length(L, 0).

my_length([], Acc) ->
    Acc;
my_length([_|T], Acc) ->
    my_length(T, Acc+1).


    

File: prolog/p-04.pl
my_length(Len, Ls) :- my_length(Len, 0, Ls).

my_length(Acc, Acc, []).
my_length(Len, Acc, [_|T]) :-
    Acc1 is Acc + 1,
    my_length(Len, Acc1, T), !.
